src/libsystemd-network/dhcp-identifier.c

```diff
@@ -21,6 +21,8 @@
 #define APPLICATION_ID SD_ID128_MAKE(a5,0a,d1,12,bf,60,45,77,a2,fb,74,1a,b1,95,5b,03)
 #define USEC_2000       ((usec_t) 946684800000000) /* 2000-01-01 00:00:00 UTC */
 
+static bool dhcp_iaid_use_mac = false;
```

Comment:
hmm, a static variable, isn't there a better way?

maybe change dhcp_identifier_set_iaid() to read some env var via secure_getenv() first, that can be used to override the string that it gets from udev?

Reply:
Hmm, this is only used by tests. I do not want to expose any option for that through env var. If it is exposed, then some one may use it for unintended way...

Another option is add a boolean argument to dhcp_identifier_set_iaid() and introduce dhcp_client_set_test_mode() or so. WDYT?

Reply:
i think that would be much nicer, yes

Reply:
OK. Thanks.